### PR TITLE
firmware: Now sending imu type based on feature

### PR DESCRIPTION
**Pull request by TheButlah**
Something is being wonky on my end, but I'm v tired. If someone could test this, I would appreciate it. It seemed to be working *before* this PR, but I'm not sure why it would change

### CODE REVIEW

File: firmware/src/main.rs [hunk 1]
```diff
@@ -67,55 +73,74 @@ fn main() -> ! {
...
-			// When heartbeat is received, we should reply with heartbeat 0 aka Discovery
-			// The protocol is asymmetric so its a bit unintuitive. TODO: Split packet type into Server2Client and C2S
-			CbPacket::Heartbeat => {
-				packets.serverbound.send(SbPacket::Heartbeat).await;
-			}
-			// Pings are basically like heartbeats, just echo data back
-			CbPacket::Ping { id } => {
-				packets.serverbound.send(SbPacket::Ping { id }).await;
-			}
+	let sensor_info_fut = async {
```
Comment (TheDevMinerTV) [inline firmware/src/main.rs:79]: this should only happen once at startup

File: firmware/src/main.rs [hunk 2]
```diff
@@ -67,55 +73,74 @@ fn main() -> ! {
...
-				sensor_id: 0,
-				data_type: 1,
-				quat: quat.wait().await.into_inner().into(),
-				calibration_info: 0,
-			})
-			.await;
+async fn do_work(packets: &Packets, quat: &Unreliable<Quat>) {
+	match packets.clientbound.recv().await {
+		// Identify ourself when discovery packet is received
+		CbPacket::Discovery => {
```
Comment (TheDevMinerTV) [inline firmware/src/main.rs:108]: if you're connected already, don't echo back

Comment (TheButlah) [reply firmware/src/main.rs:108]: how are we connected already?

Comment (TheButlah) [reply firmware/src/main.rs:108]: oh, nvm

Comment (TheButlah) [reply firmware/src/main.rs:108]: What happens if we don't know if we disconnnected or not? lets say its a new server

File: firmware/src/main.rs (hunk 1 above)
Comment (TheButlah) [reply firmware/src/main.rs:79]: at startup, or after the Discovery packet?

File: firmware/src/main.rs (hunk 2 above)
Comment (TheDevMinerTV) [reply firmware/src/main.rs:108]: just check when the last heartbeat was that you've received + some offset

File: firmware/src/main.rs (hunk 1 above)
Comment (TheDevMinerTV) [reply firmware/src/main.rs:79]: After discovery packet, or when you get an IMU error

**Comment by TheButlah** [comment]
Getting lots of: 
Screen Shot 2022-12-26 at 4 28 49 AM

**Comment by TheDevMinerTV** [comment]
the server replies back with a slimmed down sensor info packet which just contains the id and the status.
maybe you're failing to parse that

File: firmware/src/main.rs (hunk 2 above)
Comment (TheButlah) [reply firmware/src/main.rs:108]: will do later

**Pull request merged**